constraints/equalStrength.js

```diff
@@ -0,0 +1,182 @@
+/* Equal Strength Boundaries Constraints
+ * as proposed by Nick Kalivoda, August 2019:
+ * Assign violations if a phonological terminal is at the left/right edge of a
+ * different number of syntactic constituents as phonological constituents.
+ * Specific functions defined below. Base function counts number of left/right
+ * edges a terminal falls on.
+ */
+
+ //this code isn't robust for terminal order-shifting, does it need to be?
```

Comment:
I think it's okay that it's not robust to reordering of the terminals. Since we don't have a good way to refer to edges directly (they are not objects), making these functions robust to reordering would be a huge pain.

Reply:
I actually realized that this code might be robust for order-changing analyses, see the email chain "Abstract Match Constraints Update." I removed the comment to avoid further confusion, though.